index.bs

```diff
@@ -976,319 +976,352 @@ their contract with the developer.
-there is an advantage for that object to be live so that the results
-can be computed lazily, such as for {{Window/getComputedStyle()}}.
-On the other hand,
-if the data needed by an object are expensive to keep up-to-date,
-such as for the {{NodeList}} returned from {{ParentNode/querySelectorAll()}},
-then providing a static object avoids
-having to keep the object updated until it is garbage collected
-(which may be substantially after its last use).
-
-Likewise, the choice of live versus static objects
-can influence the memory use of an API.
-If each call of a method returns a new static object,
-and the objects are large,
-then substantial amounts of memory can be wasted
-until the next garbage collection.
-
-The choice of whether an object is live or static may also
-influence whether it should be returned
-from an attribute getter or from a method.
-See [[#attributes-like-data]].
-In particular,
-if a result that changes frequently is returned as a static object,
-it should probably be returned from a method rather than an attribute getter.
-
-<h3 id="prefer-dict-to-bool">Prefer dictionary parameters over boolean parameters or other unreadable parameters</h3>
-
-APIs should generally prefer dictionary parameters
-(with named booleans in the dictionary)
-over boolean parameters.
-This makes the code that calls the API
-<a href="https://ariya.io/2011/08/hall-of-api-shame-boolean-trap">much more readable</a>.
+If an API gives access to an object
+representing some internal state,
+decide whether that object should continue to be updated as the state changes.
+
+An object which represents the current state at all times
+is a <dfn export>live object</dfn>,
+while an object which represents the state at the time it was created
+is a <dfn export>static object</dfn>.
+
+<h4 id="live-objects" class="no-num no-toc">Live objects</h4>
+
+If an object allows the author to change the internal state,
+that object should be live.
+For example, DOM {{Node}}s are live objects,
+to allow the author to make changes to the document
+with an understanding of the current state.
+
+Properties of live objects may be computed as they are accessed,
+instead of when the object is created.
+This makes live objects sometimes a better choice
+if the data needed is complex to compute,
+since there is no need to compute all the data before the object is returned.
+
+A live object may also use less memory,
+since there is no need to copy data to a static version.
+
+<h4 id="static-objects" class="no-num no-toc">Static objects</h4>
+
+If an object represents a list that might change,
+most often the object should be static.
+This is so that code iterating over the list
+doesn't need to handle the possibility of the list changing in the middle.
+
+<div class="example">
+{{Document/getElementsByTagName()}} returns a live object which represents a list,
+meaning that authors need to take care when iterating over its items.
+
+<!-- does anyone know more specifics here? -->
```

Comment:
I think the specifics are in the previous sentence. If we had a time machine, `{{Document/getElementsByTagName()}}` would return a static list, because it being live means authors mess things up all the time when doing dom manipulation while iterating over its return value. `{{ParentNode/querySelectorAll()}}` was designed after we were aware of how much of a footgun `{{Document/getElementsByTagName()}}` is.

Reply:
I added an example to show how iteration goes funky when using `getElementsByTagName()`.